api/client/service/ps.go

```diff
@@ -44,13 +48,34 @@ func runPS(dockerCli *client.DockerCli, opts psOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	service, _, err := client.ServiceInspectWithRaw(ctx, opts.serviceID)
-	if err != nil {
-		return err
+	filter := opts.filter.Value()
+	filterTypes := []string{"id", "name"}
+
+	unknownServices := []string{}
+
+service_search:
+	for _, service := range opts.services {
+		for _, filterType := range filterTypes {
+			serviceFilter := filters.NewArgs()
+			serviceFilter.Add(filterType, service)
+			if serviceList, err := client.ServiceList(ctx, types.ServiceListOptions{Filter: serviceFilter}); err == nil && len(serviceList) != 0 {
+				for _, serviceEntry := range serviceList {
+					filter.Add("service", serviceEntry.ID)
+				}
+				continue service_search
```

Comment:
@stevvooe The outmost loop `for _, service := range opts.services` is to iterate through the input term(ID/Name) of the `service ps`.

The inner loop `for _, filterType := range filterTypes` is to search with `id` then with `name`. 

During the search process, if either `id` or `name` search returns anything, it will add the serviceEntry.ID to the `filter` (for process later), and continue with the next term(ID/Name) of the service search (`continue service_search`).

If both `id` and `name` search returns nothing, then it will be added to the `unknownServices`.

Previously I adds all the search term (Name/ID) of `service ps` to `id` filter first, do one search, and then to `name` filter, do another search.

However, that will not give us the information on which search term (Name/ID) is a hit and which is not.

So I just search each term (Name/ID) one by one, first through `id`, then through `name`. This will allows us to show unfound terms like `docker inspect`.